benchmarks/benchmark/tools/profile-generator/container/benchmark_serving.py

```diff
@@ -272,75 +274,126 @@ async def benchmark(
-  # dimensions values are strings
-  dimensions_json = {}
-  # metrics values are numerical
-  metrics_json = {}
-
   # Setup
-  current_dt = datetime.now().strftime("%Y%m%d-%H%M%S")
-  dimensions_json["date"] = current_dt
-  dimensions_json["backend"] = args.backend
-  dimensions_json["model_id"] = args.model
-  dimensions_json["tokenizer_id"] = args.tokenizer
-  if args.additional_metadata_metrics_to_save is not None:
-    dimensions_json = {
-        **dimensions_json,
-        **json.loads(args.additional_metadata_metrics_to_save),
+  current_dt = datetime.now()
+  current_dt_proto = Timestamp()
+  current_dt_proto.FromDatetime(current_dt)
+
+  final_json = {
+    # metrics values are numerical
+    "metrics" : {
+      # Traffic
+      "num_prompts": args.num_prompts,
+      "request_rate": args.request_rate,
+      'server_metrics': {
+        **benchmark_result["metric_names"]
+      }
+    },
+    # dimensions values are strings
+    "dimensions": {
+      "date": current_dt.strftime('%Y%m%d-%H%M%S'),
+      "backend": args.backend,
+      "model_id": args.model,
+      "tokenizer_id": args.tokenizer,
+      **(json.loads(args.additional_metadata_metrics_to_save) if args.additional_metadata_metrics_to_save else {})
+    },
+    "config": {
+      "model": args.model,
+      "model_server": args.backend,
+      "start_time": current_dt_proto.ToJsonString(),
+    },
+    "summary_stats": {
+      "stats": {
+        **benchmark_result["metric_aliases"]
+      }
     }
-  metrics_json["num_prompts"] = args.num_prompts
-
-  # Traffic
-  metrics_json["request_rate"] = args.request_rate
-  metrics_json = {**metrics_json, **benchmark_result}
-
-  final_json = {}
-  final_json["metrics"] = metrics_json
-  final_json["dimensions"] = dimensions_json
-
+  }
+  
   # Save to file
   base_model_id = args.model.split("/")[-1]
   file_name = (
-      f"{args.backend}-{args.request_rate}qps-{base_model_id}-{current_dt}.json"
+      f"{args.backend}-{args.request_rate}qps-{base_model_id}-{current_dt.strftime('%Y%m%d-%H%M%S')}.json"
   )
   with open(file_name, "w", encoding="utf-8") as outfile:
     json.dump(final_json, outfile)
 
-def metrics_to_scrape(backend: str) -> List[str]:
-  if backend == "vllm":
-    return ["vllm:gpu_cache_usage_perc", "vllm:num_requests_waiting"]
-  elif backend == "jetstream":
-    return ["jetstream_slots_used_percentage", "jetstream_prefill_backlog_size"]
-  else:
-    return []
-
-def print_metrics(metrics: List[str], duration: float, backend: str):
+def metrics_to_scrape(backend: str) -> Dict[str, Optional[str]]:
+    # Each key in the map is a metric, it has a corresponding 'stats' object
+    # It must be populated on the outputs 'metrics' field as 'key':'stats'
+    # If a value is specified for a given key, it will be populated on the outputs `summary_stats.stats` field as 'value':'stats' as well.
+    if backend == "vllm":
+        return {
+            "vllm:gpu_cache_usage_perc": None,
```

Comment:
How does vllm get ttft, tpot, request_latency if jetstream does it like below?

Reply:
Currently we don't formally report stats on these metrics, although we do collect and report [these stats](https://github.com/GoogleCloudPlatform/ai-on-gke/pull/832/files#diff-99d0648ee83e9dbc3c69f01cf7bf87667772caf5b259b0bcd5d5451268d1793cR466-R535). Ideally we can collect these stats from prometheus, this will be addressed in a followup.

Reply:
So, where do we have the data the benchmark produced in the report? We report request latency, average time per output token, etc.